### PR TITLE
feat: support post_logout_redirect_uri config in openid-connect plugin

### PR DESCRIPTION
**Pull request by starsz**
### What this PR does / why we need it:

Hi, this PR is used to support post_logout_redirect_uri config in openid-connect plugin.
So that we can redirect the uri when request the logout path of openid-connect plugin.
Fix: https://github.com/apache/apisix/issues/6345, https://github.com/apache/apisix/issues/6362

Since the lua-resty-openidc has support post_logout_redirect_uri already, we can just add post_logout_redirect_uri  config in our plugin.
Refer: https://github.com/zmartzone/lua-resty-openidc/blob/3aac462f8206b8553cd1f34bb99b1788efbc2140/lib/resty/openidc.lua#L1331-L1339

### Pre-submission checklist:

* [x] Did you explain what problem does this PR solve? Or what new features have been added?
* [x] Have you added corresponding test cases?
* [x] Have you modified the corresponding document?
* [x] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix/tree/master#community) first**

### CODE REVIEW

File: docs/en/latest/plugins/openid-connect.md [hunk 1]
```diff
@@ -47,6 +47,7 @@ The OAuth 2 / Open ID Connect(OIDC) plugin provides authentication and introspec
 | realm                                | string  | optional    | "apisix"              |         | Realm used for the authentication                                                                                               |
 | bearer_only                          | boolean | optional    | false                 |         | Setting this `true` will check for the authorization header in the request with a bearer token                                  |
 | logout_path                          | string  | optional    | "/logout"             |         |                                                                                                                                 |
+| post_logout_redirect_uri   | string | optional |              |  | URL want to redirect when request logout_path
```
Comment (spacewander) [inline docs/en/latest/plugins/openid-connect.md:50]: please align the table

File: t/plugin/openid-connect.t [hunk 2]
```diff
@@ -1665,3 +1665,319 @@ GET /t
...
+                        "upstream": {
+                            "nodes": {
+                                "127.0.0.1:1980": 1
+                            },
+                            "type": "roundrobin"
+                        },
+                        "uri": "/*"
+                }]],
+                [[{
+                    "node": {
```
Comment (spacewander) [inline t/plugin/openid-connect.t:1706]: We don't need to check the response data of this request, right?

File: t/plugin/openid-connect.t [hunk 3]
```diff
@@ -1665,3 +1665,319 @@ GET /t
...
+                        local len = #cookies
+                        if len > 0 then
+                            cookie_str = cookies[1]:match('([^;]*); .*')
+                            for i = 2, len do
+                                cookie_str = cookie_str .. "; " .. cookies[i]:match('([^;]*); .*')
+                            end
+                        end
+                    end
+
+                    -- Get the final URI out of the Location response header. This should be the original URI that was requested.
```
Comment (spacewander) [inline t/plugin/openid-connect.t:1916]: The similar Lua snipper is repeated in multiple tests. Could we refactor it?

File: t/plugin/openid-connect.t (hunk 2 above)
Comment (starsz) [reply t/plugin/openid-connect.t:1706]: I think we should check the response data of this request like other route-creating tests.
Why we don't need to check the response?

File: docs/en/latest/plugins/openid-connect.md (hunk 1 above)
Comment (starsz) [reply docs/en/latest/plugins/openid-connect.md:50]: OK. Let me fix it.

File: t/plugin/openid-connect.t (hunk 2 above)
Comment (spacewander) [reply t/plugin/openid-connect.t:1706]: Yes, we check the response status code but not the data. The data is mostly an echo of the input.

File: t/plugin/openid-connect.t (hunk 3 above)
Comment (spacewander) [reply t/plugin/openid-connect.t:1916]: https://github.com/apache/apisix/blob/acbb111afbbd096444bdd6dc0307d18187b19829/t/plugin/skywalking.t#L30
You can define helper functions in extra_init_by_lua

Comment (starsz) [reply t/plugin/openid-connect.t]: Fixed. Please review it again.

File: t/plugin/openid-connect.t (hunk 2 above)
Comment (starsz) [reply t/plugin/openid-connect.t:1406]: Let me remove it.

File: t/plugin/openid-connect.t (hunk 3 above)
Comment (starsz) [reply t/plugin/openid-connect.t]: Fixed. Please review it again.

**Review by tzssangglass — APPROVED** [review]

**Review by spacewander — APPROVED** [review]

**Pull request merged**